### PR TITLE
feat: enable resolve_root_symlink by default

### PR DESCRIPTION
**Pull request by dunglas**
Not having this by default seems to break Laravel apps using `public/storage/`.

As you know well this part and Laravel, would you mind to review this @francislavoie? Thank you!

### CODE REVIEW

File: caddy/caddy.go [hunk 1]
```diff
@@ -25,7 +25,7 @@ const defaultDocumentRoot = "public"
 
 func init() {
 	caddy.RegisterModule(FrankenPHPApp{})
-	caddy.RegisterModule(FrankenPHPModule{})
+	caddy.RegisterModule(FrankenPHPModule{ResolveRootSymlink: true})
```
Comment (francislavoie) [inline caddy/caddy.go:28]: I don't think this is right. Shouldn't pass options here.

The typical way to have a boolean that you can detect not having been specified by the user is by using a `*bool` instead so the default is actually nil.

Then you can set the value in Provision based on whether it was user-specified or not.

File: docs/config.md [hunk 2]
```diff
@@ -122,7 +122,7 @@ The `php_server` and the `php` directives have the following options:
 php_server [<matcher>] {
 	root <directory> # Sets the root folder to the site. Default: `root` directive.
 	split_path <delim...> # Sets the substrings for splitting the URI into two parts. The first matching substring will be used to split the "path info" from the path. The first piece is suffixed with the matching substring and will be assumed as the actual resource (CGI script) name. The second piece will be set to PATH_INFO for the CGI script to use. Default: `.php`
-	resolve_root_symlink # Enables resolving the `root` directory to its actual value by evaluating a symbolic link, if one exists.
+	resolve_root_symlink false # Disableds resolving the `root` directory to its actual value by evaluating a symbolic link, if one exists (enabled by default).
```
Comment (francislavoie) [inline docs/config.md:125]:
```suggestion
	resolve_root_symlink false # Disables resolving the `root` directory to its actual value by evaluating a symbolic link, if one exists (enabled by default).
```

**Comment by francislavoie** [comment]
The reason we didn't enable it by default is to avoid a tiny performance hit from the extra syscall which may be wasteful if the root is never a symlink.

Felt better to default to having the best performance rather than best compat because it's not everyone that will use symlinks as a deployment strategy and if they do it's trivial to enable.

I have no issue with this being on by default for FrankenPHP.

File: caddy/caddy.go (hunk 1 above)
Comment (dunglas) [reply caddy/caddy.go]: Fixed, thanks!

**Comment by dunglas** [comment]
@francislavoie ok! Let's have better compatibility with Laravel and the like (probably helps with Symfony too) and let users create a custom Caddyfile for performance optimizations.

**Review by francislavoie — APPROVED** [review]

**Pull request merged**